packages/collector/test/tracing/messaging/nats/test.js

```diff
@@ -6,320 +6,481 @@
-                    // listener. By design, we deliberately do not register our own event listener on nats, so publisher
-                    // errors are not captured.
-                    expectations.push(span => expect(span.ec).to.equal(1));
-                    expectations.push(span => expect(span.data.nats.error).to.equal('Subject must be supplied'));
-                  } else {
-                    expectations.push(span => expect(span.ec).to.equal(0));
-                    expectations.push(span => expect(span.data.nats.error).to.not.exist);
-                  }
-
-                  if (!withError) {
-                    // we omit the subject to trigger an error, that's why we only test it in non-error tests
-                    expectations.push(span => expect(span.data.nats.subject).to.equal('publish-test-subject'));
-                  }
-
-                  expectExactlyOneMatching(spans, expectations);
-
-                  // verify that subsequent calls are correctly traced
-                  expectAtLeastOneMatching(spans, [
-                    span => expect(span.n).to.equal('node.http.client'),
-                    span => expect(span.t).to.equal(entrySpan.t),
-                    span => expect(span.p).to.equal(entrySpan.s),
-                    span => expect(span.k).to.equal(constants.EXIT)
+                    span => expect(span.data.nats.sort).to.equal('consume'),
+                    span => expect(span.data.nats.subject).to.equal('publish-test-subject'),
+                    span =>
+                      version === 'v1'
+                        ? expect(span.data.nats.address).to.equal('nats://localhost:4222')
+                        : expect(span.data.nats.address).to.equal('nats://127.0.0.1:4222'),
+                    span => expect(span.data.nats.error).to.not.exist
                   ]);
-                });
+                }
               });
+          });
+
+          if (version === 'latest') {
+            it('should not trace publish when suppressed without headers', async function () {
+              return publisherControls
+                .sendRequest({
+                  method: 'POST',
+                  path: '/publish?noHeaders=true',
+                  suppressTracing: true
+                })
+                .then(res => {
+                  expect(res).to.equal('OK');
+                  return delay(config.getTestTimeout() / 4);
+                })
+                .then(() => agentControls.getSpans())
+                .then(spans => {
+                  expect(spans).to.have.lengthOf(0);
+                });
             });
-        }
-      );
-    }
-  });
+          }
+        });
+      });
 
-  describe('subscribe', function () {
-    [false, true].forEach(withError => {
-      testSubscribe.call(this, withError);
-    });
+      // NOTE: We do not track these errors yet, because there is no entry span
+      describe.skip('connect', () => {
```

Comment:
@basti1302 Never noticed that we do not trace connection errors for instrumentations. Is that something we want to trace in the future?

Reply:
Hm, well, as you wrote: Usually this happens outside of active tracing, in the startup phase of the application, before it even starts to process incoming requests. The general rule "we don't trace an exit if there is no active entry span" still holds. So specifically for NATS I think we are fine without tracing this? 

If, on the other hand, we would instrument a driver that typically establishes a connection while processing an incoming request, we would probably want to instrument that as well.

Reply:
> The general rule "we don't trace an exit if there is no active entry span" still holds.

Yes. The question is: Is that something we want to trace in the future in general for all instrumentations (independent of the engine)?

Reply:
AFAIK there are no plans to change this rule.